training/src/anemoi/training/diagnostics/callbacks/plot.py

```diff
@@ -824,6 +828,24 @@ def automatically_determine_group(name: str) -> str:
             legend_patches,
         )
 
+    def argsort_name_variablelevel(self) -> list[int]:
+        """Custom sort key to process the strings.
+
+        Sort parameter names by alpha part, then by numeric part at last
+        position (variable level), then by the original string.
+        """
+        data = self.parameter_names
+
+        def custom_sort_key(index: int) -> tuple:
+            s = data[index]  # Access the element by index
+            parts = s.split("_")
+            alpha_part = s if len(parts) == 1 else s[: -len(parts[-1]) - 1]
+            numeric_part = int(parts[-1]) if len(parts) > 1 and parts[-1].isdigit() else float("inf")
+            return (alpha_part, numeric_part, s)
```

Comment:
Why do you need to return s here too?

Reply:
If `alpha_part` and `numeric part` are the same for two different variables, the entire variable name `s` is used for sorting.